core/src/sumeragi/main_loop.rs

```diff
@@ -1325,3 +1294,55 @@ where
         }
     }
 }
+
+/// Delete elements from vector for which `p(e) == false`, preserving order of all other elements.
+fn truncate_false<T>(v: &mut Vec<T>, p: impl Fn(&T) -> bool) {
```

Comment:
Isn't this function's purpose pretty much what `Vec::retain` does?

Reply:
Good catch, thanks, i overlooked this method.